base/broadcast.jl

```diff
@@ -500,14 +500,13 @@ end
             @nexprs $nargs i->(@inbounds val_i = _broadcast_getindex(A_i, I_i))
             # call the function
             V = @ncall $nargs f val
-            S = typeof(V)
             # store the result
-            if S <: eltype(B)
+            if V isa eltype(B)
```

Comment:
+1

This syntax allows the compiler to see / "prove" that `S <: T` if this branch is taken, and thus likely eliminate dispatch in the assignment below